elliottlib/errata_async.py

```diff
@@ -0,0 +1,116 @@
+import asyncio
+import base64
+import logging
+from typing import Dict, List, Set, Union
+from urllib.parse import quote, urlparse
+
+import aiohttp
+import gssapi
+from kobo.rpmlib import parse_nvr
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AsyncErrataAPI:
+    def __init__(self, url: str):
+        self._errata_url = urlparse(url).geturl()
+        self._session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(limit=32))
+        self._gssapi_client_ctx = None
+        self._headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        }
+
+    async def login(self):
+        if self._gssapi_client_ctx:
+            return  # already logged in
+        server_name = gssapi.Name(f"HTTP/{urlparse(self._errata_url).hostname}")
+        client_ctx = gssapi.SecurityContext(name=server_name, usage='initiate')
+        out_token = client_ctx.step(b"")
+        self._gssapi_client_ctx = client_ctx
+        self._headers["Authorization"] = 'Negotiate ' + base64.b64encode(out_token).decode()
+
+    async def close(self):
+        await self._session.close()
+
+    async def _make_request(self, method: str, path: str, parse_json: bool = True, **kwargs) -> Union[Dict, bytes]:
+        if "headers" not in kwargs:
+            kwargs["headers"] = self._headers
+        async with self._session.request(method, self._errata_url + path, **kwargs) as resp:
+            resp.raise_for_status()
+            result = await (resp.json() if parse_json else resp.read())
+        return result
+
+    async def get_advisory(self, advisory: Union[int, str]) -> Dict:
+        path = f"/api/v1/erratum/{quote(str(advisory))}"
```

Comment:
Could paths like these be consumed from `elliottlib/constants.py`?

Reply:
The API endpoint string is used in more than one place, how about moving its definition at module level?